compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1698,14 +1698,33 @@ convertDataCon env m con args
     | Just (tyArgs, tmArgs) <- splitConArgs_maybe con args = do
         tyArgs <- mapM (convertType env) tyArgs
         tmArgs <- mapM (convertExpr env) tmArgs
+        (, []) <$> fullyApplied env tyArgs tmArgs
+
+    -- Partially applied, but the constructor only takes type args.
+    -- In this situation there is no worker function, so we inline
+    -- the constructor by introducing some type lambdas.
+    | let (conTypes, conTheta, conArgs, _) = dataConSig con
+    , length conTheta + length conArgs == 0 -- no value args
```

Comment:
```suggestion
    , null conTheta && null conArgs -- no value args
```
maybe a bit clearer? (and potentially slightly more efficient)

Reply:
ah, nice, yes :)